src/Scrutor/ServiceCollectionExtensions.Composite.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static partial class ServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Registers <typeparamref name="TConcrete"/> as a composite that wraps all
+        /// existing registrations of <typeparamref name="TInterface"/>.
+        /// </summary>
+        /// <typeparam name="TInterface"></typeparam>
+        /// <typeparam name="TConcrete"></typeparam>
+        /// <param name="services"></param>
+        public static void AddComposite<TInterface, TConcrete>(this IServiceCollection services)
+          where TInterface : class
+          where TConcrete : class, TInterface
+        {
+            var wrappedDescriptors = services.Where(s => s.ServiceType == typeof(TInterface)).ToList();
```

Comment:
Should this be `typeof(TInterface).IsAssignableFrom(s.ServiceType)`?

Reply:
Good call - done